pkg/asset/agent/image/agentimage.go

```diff
@@ -129,6 +139,78 @@ func (a *AgentImage) updateIgnitionImg(ignition []byte) error {
 	return nil
 }
 
+func (a *AgentImage) updateCdBootImg(ignition []byte) error {
+
+	type IgnInfo struct {
+		File   string `json:"file"`
+		Length int64  `json:"length"`
+		Offset int64  `json:"offset"`
+	}
+	var ignInfo IgnInfo
+
+	// Reading the ignition details from the json file
+	ignInfoJSONPath := filepath.Join(a.tmpPath, "coreos", "igninfo.json")
```

Comment:
I think that this is a more generic mechanism, and it should be used always (regardless of the current architecture) to detect the archive name that will contain the ignition file  (and defaulted to `ignition.img` if not found). 
I'd expect that it would be sufficient to inject such logic into [updateIgnitionImg](https://github.com/openshift/installer/blob/e2d6013693b4f7b9fa4bbe3b1cc1be11f997f722/pkg/asset/agent/image/agentimage.go#L106), thus eliminating all the current duplication and keeping just one method